lib/event_listeners.js

```diff
@@ -92,7 +92,7 @@ AWS.EventListeners = {
     addAsync('COMPUTE_SHA256', 'afterBuild', function COMPUTE_SHA256(req, done) {
       req.haltHandlersOnError();
       if (!req.service.api.signatureVersion) return done(); // none
-      if (req.service.getSignerClass(req) === AWS.Signers.V4) {
+      if (req.service.getSignerClass(req.isPresigned()) === AWS.Signers.V4) {
```

Comment:
Keeping the argument as `req` instead of a boolean indicating if it's presigned would keep this file more generic to all services, since determining signer class based on if the request is presigned is S3 specific. Although the generic service `getSignerClass` currently does not use any arguments, it's possible that more custom `getSignerClass` methods could be written for individual services. The `req.isPresigned()` could easily be moved into S3's custom `getSignerClass`.